### PR TITLE
feat(client): fetch get-session-user without checking jwt

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

This PR makes it so the client always tries to fetch the session user (even if the `jwt` cookie is missing).  This is to allow the client to use the new api, since the new api sets httpOnly session cookies that cannot be inspected by the client.

Without these changes the client will never appear to be logged in, even if it has authenticated.

### CODE REVIEW

**Review by moT01 — APPROVED** [review]
It works as intended - so LGTM if we want it :tada: 

I do worry that it will make for a lot of extra requests - not sure how many though. Is this temporary or will it be replaced with something else at some point @ojeytonwilliams? Seems like if we can eliminate extra requests, it's probably a good idea.

**Review by raisedadead — COMMENTED** [review]
Elaborate the change or add/expand tests.

The OP description feels like walking into the middle of a conversation, and I would not remember the context a few weeks after this has landed.

**Comment by ojeytonwilliams** [comment]
The context is that we are planning to use a httpOnly session cookie in the new api, so the client cannot check that cookie before requesting the session user. Simply making the request regardless seemed like the simplest solution.

I'm not sure how best to test this, but I'll try to come up with something.

**Comment by raisedadead** [comment]
> I'm not sure how best to test this, but I'll try to come up with something.

That's fine; at least add the context somewhere better than a comment. 

Plainly speaking, the PR touches code that we use in production if merged. If you ask someone to read the OP or commit a message, they will have zero idea of what you are talking about.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**
